jupyterlab_git/git.py

```diff
@@ -1112,6 +1112,48 @@ def remote_add(self, top_repo_path, url, name=DEFAULT_REMOTE_NAME):
                 "message": my_error.decode("utf-8").strip()
             }
 
+    async def ensure_gitignore(self, top_repo_path):
+        """Handle call to ensure .gitignore file exists and the 
+        next append will be on a new line (this means an empty file
+        or a file ending with \n).
+
+        top_repo_path: str
+            Top Git repository path
+        """
+        try:            
+            gitignore = os.path.join(top_repo_path, ".gitignore")
+            if not os.path.exists(gitignore):
+                open(gitignore, "w").close()
+            elif os.path.getsize(gitignore) != 0:
+                f = open(gitignore, "r")
+                if (f.read()[-1] != "\n"):
+                    f = open(gitignore, "a")
+                    f.write('\n')
+                f.close()
+            return {"code": 0}
+        except:
+            return {"code": -1}
+
+    async def ignore(self, top_repo_path, file_path):
+        """Handle call to add an entry in .gitignore.
+
+        top_repo_path: str
+            Top Git repository path
+        file_path: str
+            The path of the file in .gitignore 
+        """
+        try:
+            res = await self.ensure_gitignore(top_repo_path)
+            if res["code"] != 0:
+                return res
+            gitignore = os.path.join(top_repo_path, '.gitignore')
+            f = open(gitignore, "a")
+            f.write(file_path + "\n")
+            f.close()
+            return {"code": 0}
+        except:
+            return {"code": -1}
```

Comment:
```suggestion
            gitignore = pathlib.Path(top_repo_path) / ".gitignore"
            with gitignore.open("a") as f:
                f.write(file_path + "\n")
        except BaseException as error:
            return {"code": -1, "message": str(error)}}
        else:
            return {"code": 0}
```

Reply:
Thx for the review. Last commit implements this.